pkg/compactor/block_upload_test.go

```diff
@@ -1372,3 +1348,107 @@ func TestMultitenantCompactor_GetBlockUploadStateHandler(t *testing.T) {
 		})
 	}
 }
+
+func TestMultitenantCompactor_FinishBlockUpload(t *testing.T) {
```

Comment:
(Rant, ignore) This test and other tests in this file checks that `parseBlockUploadParameters` works. Ideally we would tests that only once, not again and again in each handler.